includes/admin/admin-actions.php

```diff
@@ -403,3 +403,58 @@ function give_blank_slate() {
 	$blank_slate->init();
 }
 add_action( 'current_screen', 'give_blank_slate' );
+
+/**
+ * Validate Fields of User Profile
+ *
+ * @param object   $errors Object of WP Errors.
+ * @param int|bool $update True or False.
+ * @param object   $user   WP User Data.
+ *
+ * @since 2.0
+ *
+ * @return mixed
+ */
+function give_validate_user_profile( $errors, $update, $user ) {
+
+	if ( ! empty( $_POST['action'] ) && ( 'adduser' === $_POST['action'] || 'createuser' === $_POST["action"] ) ) {
+		return;
+	}
+
+	if( ! empty( $user->ID ) ) {
+		$donor = Give()->donors->get_donor_by( 'user_id', $user->ID );
+
+		if( $donor ) {
+			// If Donor is attached with User, then validate first name
+			if ( empty( $_POST['first_name'] ) ) {
+				$errors->add('empty_first_name', '<strong>ERROR:</strong>' . __( 'Please enter your first name.', 'give' ) );
+			}
+		}
+	}
+
+}
+
+add_action( 'user_profile_update_errors', 'give_validate_user_profile', 10, 3 );
+
+/**
+ * @param $user
+ */
+function give_donor_information_profile_fields( $user ) {
+	$donor = Give()->donors->get_donor_by( 'user_id', $user->ID );
```

Comment:
@mehul0810 Do not show link if user does not connect to donor

Reply:
@ravinderk Updated Code to only show Donor Information is Donor and User are attached.